src/device_constructors/storage_constructor.jl

```diff
@@ -0,0 +1,83 @@
+function construct_device!(ps_m::CanonicalModel,
+                           category::Type{St},
+                           category_formulation::Type{D},
+                           system_formulation::Type{S},
+                           sys::PSY.PowerSystem;
+                           kwargs...) where {St <: PSY.Storage,
+                                             D <: PSI.AbstractStorageForm,
+                                             S <: PM.AbstractPowerFormulation}
+
+    #wrangle initial_conditions
+    if  !isempty(keys(ps_m.initial_conditions))
+
+        "energy_initial_conditions" in keys(ps_m.initial_conditions) ? energy_initial_conditions = ps_m.initial_conditions["energy_initial_conditions"] : @warn("No energy initial conditions provided")
```

Comment:
There is a common pattern that has a bug (I mentioned it yesterday, but adding a comment as a reminder).   An UndefVarError exception will get thrown if ps_m.initial_conditions is not empty but does not have "energy_initial_conditions".  Suggest writing the code this way:

energy_initial_conditions = get(ps_m.initial_conditions, "energy_initial_conditions", zeros(length(sys.storage)))